cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -301,9 +301,14 @@ public void performRunLevelStop(HttpRequest request, HttpResponder responder,
                                   @PathParam("program-type") String type,
                                   @PathParam("program-id") String programId,
                                   @PathParam("run-id") String runId,
-                                  @QueryParam("graceful") String gracefulShutdownSecs) throws Exception {
+                                  @QueryParam("graceful") String gracefulShutdownSecs,
+                                  @QueryParam("version") String version) throws Exception {
     ProgramType programType = getProgramType(type);
-    ProgramId program = new ProgramId(namespaceId, appId, programType, programId);
+    ProgramId program = (version == null) ?
```

Comment:
nit, to increase readability a bit: 
```
    if (version == null) {
      version = getLatestAppVersion(new NamespaceId(namespaceId), appId);
    }
    ProgramId program = new ApplicationId(namespaceId, appId, version).program(programType, programId);
```

Reply:
Also, pls update the javadoc

Reply:
Thanks for the suggestion. I will probably come back for this after this [PR](https://github.com/cdapio/cdap/pull/14605) is merged, because we won't need version here, runId is all we need

Reply:
cool, I'm also working on addressing the PR comments